### ISSUE TITLE
Example dataset run gives errors with R 4.4.2

### ISSUE THREAD

**Issue by mclaughj23** [selcukorkmaz/fastml#5]
Hello.  Working to understand fastml and ran example dataset.  Following results and not sure how to fix.  Thank you. -John
--------------------------------
> library(fastml)
> 
> # Example dataset
> data(iris)
> iris <- iris[iris$Species != "setosa", ]  # Binary classification
> iris$Species <- factor(iris$Species)
> 
> # Train models
> model <- fastml(data = iris,label = "Species")
Error in `add_model()`:
! parsnip could not locate an implementation for `boost_tree` classification model specifications using
  the `lightgbm` engine.
ℹ The parsnip extension package bonsai implements support for this specification.
ℹ Please install (if needed) and load to continue.

Run `rlang::last_trace()` to see where the error occurred.
Warning messages:
1: glm.fit: algorithm did not converge 
2: glm.fit: fitted probabilities numerically 0 or 1 occurred 
3: In value[[3L]](cond) :
  Training failed for algorithm: penalized_logistic_regression 
Error message: This engine requires some package installs: 'glmnet'
4: In value[[3L]](cond) : Training failed for algorithm: c5.0 
Error message: This engine requires some package installs: 'C50'
5: In value[[3L]](cond) : Training failed for algorithm: random_forest 
Error message: This engine requires some package installs: 'ranger'
6: In value[[3L]](cond) : Training failed for algorithm: ranger 
Error message: This engine requires some package installs: 'ranger'
7: In value[[3L]](cond) : Training failed for algorithm: xgboost 
Error message: This engine requires some package installs: 'xgboost'
> rlang::last_trace()
<error/rlang_error>
Error in `add_model()`:
! parsnip could not locate an implementation for `boost_tree` classification model specifications using
  the `lightgbm` engine.
ℹ The parsnip extension package bonsai implements support for this specification.
ℹ Please install (if needed) and load to continue.

-------------------------------------------------------------

**Comment by malmistry** [comment]
Tried to install all dependent R packages separately, but still gives the same error.

Tried installing both CRAN and dev versions of fastml

>sessionInfo()
R version 4.4.1 (2024-06-14)
Platform: aarch64-apple-darwin20
Running under: macOS 15.0.1

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/London
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods  
[7] base     

other attached packages:
[1] fastml_0.2.0

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.1    timeDate_4041.110   libcoin_1.0-10     
 [4] dplyr_1.1.4         digest_0.6.37       rpart_4.1.23       
 [7] timechange_0.3.0    lifecycle_1.0.4     yardstick_1.3.1    
[10] survival_3.7-0      Cubist_0.4.4        lightgbm_4.5.0     
[13] magrittr_2.0.3      compiler_4.4.1      rlang_1.1.4        
[16] tools_4.4.1         partykit_1.2-23     utf8_1.2.4         
[19] C50_0.1.8           data.table_1.16.4   xgboost_1.7.8.1    
[22] plyr_1.8.9          DiceDesign_1.10     parsnip_1.2.1      
[25] withr_3.0.2         purrr_1.0.2         workflows_1.1.4    
[28] nnet_7.3-19         grid_4.4.1          tune_1.2.1         
[31] fansi_1.0.6         colorspace_2.1-1    future_1.34.0      
[34] ggplot2_3.5.1       globals_0.16.3      scales_1.3.0       
[37] iterators_1.0.14    MASS_7.3-61         cli_3.6.3          
[40] mvtnorm_1.3-2       inum_1.0-5          generics_0.1.3     
[43] rstudioapi_0.17.1   future.apply_1.11.3 reshape2_1.4.4     
[46] stringr_1.5.1       splines_4.4.1       dials_1.3.0        
[49] parallel_4.4.1      vctrs_0.6.5         hardhat_1.4.0      
[52] glmnet_4.1-8        Matrix_1.7-1        jsonlite_1.8.9     
[55] Formula_1.2-5       listenv_0.9.1       foreach_1.5.2      
[58] gower_1.0.1         tidyr_1.3.1         recipes_1.1.0      
[61] glue_1.8.0          parallelly_1.40.1   codetools_0.2-20   
[64] rsample_1.2.1       lubridate_1.9.4     stringi_1.8.4      
[67] gtable_0.3.6        shape_1.4.6.1       munsell_0.5.1      
[70] GPfit_1.0-8         tibble_3.2.1        furrr_0.3.1        
[73] pillar_1.9.0        ipred_0.9-15        lava_1.8.0         
[76] R6_2.5.1            lhs_1.2.0           doParallel_1.0.17  
[79] lattice_0.22-6      class_7.3-22        Rcpp_1.0.13-1      
[82] prodlim_2024.06.25  ranger_0.17.0       pkgconfig_2.0.3

**Comment by selcukorkmaz** [comment]
The issue now should be fixed in the development version. Please check.

**Comment by selcukorkmaz** [comment]
Closing this issue, since it has been resolved in version 0.4.0.

**Comment by selcukorkmaz** [comment]
Closing this issue, since it has been resolved in version 0.4.0.
